src/flamegraph/flamegraph.js

```diff
@@ -158,29 +205,29 @@ function zoom_child(e, x, ratio) {
     if (e.attributes != undefined) {
         if (e.attributes.x != undefined) {
             orig_save(e, "x");
-            e.attributes.x.value = (parseFloat(e.attributes.x.value) - x - xpad) * ratio + xpad;
+            e.attributes.x.value = format_percent((parseFloat(e.attributes.x.value) - x) * ratio);
```

Comment:
Looks like `xpad` has been removed in a bunch of places. What's the reasoning behind that?

Reply:
I wanted to keep the left and right padding a fixed number of pixels when resizing. Instead of updating the positions of all of the rectangles each time, I only update the position and width of the parent element and the rectangles no longer need to care about the padding.

In order to do this, I had to change the "frames" element from a `g` to a nested `svg`.

Reply:
Ah, I see. `xpad` was really just a global x padding, not a "per element" thing.